paddle/memory/memcpy.cc

```diff
@@ -29,6 +29,16 @@ void Copy<platform::CPUPlace, platform::CPUPlace>(platform::CPUPlace, void* dst,
 }
 
 #ifndef PADDLE_ONLY_CPU
+
+template <>
+void Copy<platform::CPUPlace, platform::GPUPlace>(platform::CPUPlace dst_place,
+                                                  void* dst,
+                                                  platform::GPUPlace src_place,
+                                                  const void* src, size_t num) {
+  platform::SetDeviceId(src_place.device);
+  platform::GpuMemcpySync(dst, src, num, cudaMemcpyDeviceToHost);
```

Comment:
Maybe we do not need a sync Copy here. Copy work on a specific cuda stream too. If we really want to sync the copy:
```
Copy(dts_place, dst, src_place, src, num, stream_);
cudaStreamSynchronize(stream_);
```

At now, we only have default stream(and I am fixing it in #3497 ), and you can pass 0  as cuda stream at now.